### PR TITLE
i-bem: setMod/hasMod convert non-boolean values to string

### LINKED ISSUE #890: i-bem: setMod should convert non-boolean value to string
``` js
block.setMod('blah', 2);
console.log(typeof block.getMod('blah'));  // number

block.setMod('blah', '2');
console.log(typeof block.getMod('blah'));  // string
```

Current behaviour looks weird and it seems that any value in the example above may fire several onSetMod-handlers (see [buildModFnName()](https://github.com/bem/bem-core/blob/6ce978f653cd84a7a490e654d6c6dbf378cb7546/common.blocks/i-bem/i-bem.vanilla.js#L53-L59)).

And the same goes to `hasMod`

### PR DESCRIPTION
**Pull request by narqo**
closes #890

/cc @dfilatov

### CODE REVIEW

**Comment by dfilatov** [comment]
commit title doesn't correspond its content

**Comment by dfilatov** [comment]
I have two points about that:
- jsdoc said that it took `String` param. So passing `Number` to it can't be treated as a bug.
- we have only special case —`Boolean`, so we should explicitly cast to `String` any other values, not just `Number`.

**Comment by dfilatov** [comment]
@narqo ping

**Comment by narqo** [comment]
> jsdoc said that it took String param. So passing Number to it can't be treated as a bug.

Agree with it. But at the same time I'd prefer to have a warnings (or an error) about such things. Currently it not that obvious, how [that code works](http://jsbin.com/jobal/3/watch?html,js,console).

> [..] so we should explicitly cast to String any other values, not just Number.

I think I haven't understood your point. Which other types in addition to String and Number do you mean? 

BTW, we've fixed jsdoc of these methods as well.

**Comment by dfilatov** [comment]
> I think I haven't understood your point. Which other types in addition to String and Number do you mean?

I mean we don't need to explicitly consider `Number` as a special case. `Boolean` (and maybe `undefined`) is only special case. For any other values we should just call `.toString()`.

**Comment by qfox** [comment]
Propose `{Boolean|*}` for type (or `{?}` if we want to do here #893). And description: `true or undefined would be treated as "true" string`.
But what about false?

**Comment by dfilatov** [comment]
No, `undefined` should not be treated as `true`, especially as string, `setMod(modName, undefined)` is equivalent of deleting this mod. 
In case of `setMod(modName)` modVal should be treated as Boolean `true`.

**Comment by qfox** [comment]
@dfilatov Sorry, I meant #892

**Comment by dfilatov** [comment]
@zxqfox Do you ever sleep? )

**Comment by qfox** [comment]
@dfilatov Should I? ;-)

**Comment by narqo** [comment]
:up:

**Comment by narqo** [comment]
@dfilatov pong

File: common.blocks/i-bem/i-bem.vanilla.js [hunk 1]
```diff
@@ -283,7 +283,12 @@ var BEM = inherit(events.Emitter, /** @lends BEM.prototype */ {
             }
         }
 
-        var res = this.getMod(elem, modName) === modVal;
+        var typeModVal = typeof modVal;
+        typeModVal === 'string' ||
+            typeModVal === 'boolean' ||
+            typeModVal === 'undefined' || (modVal = modVal.toString());
+
+        var res = this.getMod(elem, modName) ===  modVal;
```
Comment (dfilatov) [inline common.blocks/i-bem/i-bem.vanilla.js]: double space before `modVal`

**Comment by dfilatov** [comment]
lgtm

**Comment by dfilatov** [comment]
don't forget about v3, will you?

**Pull request merged**